In [53]:
import kagglehub

# Download latest version
path = kagglehub.dataset_download("shriyashjagtap/e-commerce-customer-for-behavior-analysis")

print("Path to dataset files:", path)

Path to dataset files: /root/.cache/kagglehub/datasets/shriyashjagtap/e-commerce-customer-for-behavior-analysis/versions/4


In [54]:
import os

dataset_path = '/root/.cache/kagglehub/datasets/shriyashjagtap/e-commerce-customer-for-behavior-analysis/versions/4'
files = os.listdir(dataset_path)
print('Nama Files = ', files)

Nama Files =  ['ecommerce_customer_data_large.csv', 'ecommerce_customer_data_custom_ratios.csv']


In [55]:
import pandas as pd
data_file = os.path.join(dataset_path, 'ecommerce_customer_data_large.csv')
data = pd.read_csv(data_file)

In [56]:
data

Customer ID        Purchase Date Product Category  Product Price  \
0             44605  2023-05-03 21:30:02             Home            177   
1             44605  2021-05-16 13:57:44      Electronics            174   
2             44605  2020-07-13 06:16:57            Books            413   
3             44605  2023-01-17 13:14:36      Electronics            396   
4             44605  2021-05-01 11:29:27            Books            259   
...             ...                  ...              ...            ...   
249995        33807  2023-01-24 12:32:18             Home            436   
249996        20455  2021-06-04 05:45:25      Electronics            233   
249997        28055  2022-11-10 17:11:57      Electronics            441   
249998        15023  2021-06-27 14:42:12      Electronics             44   
249999         4148  2020-09-07 05:12:19             Home            307   

        Quantity  Total Purchase Amount Payment Method  Customer Age  Returns  \
0              1                   2427         PayPal            31      1.0   
1              3                   2448         PayPal            31      1.0   
2              1                   2345    Credit Card            31      1.0   
3              3                    937           Cash            31      0.0   
4              4                   2598         PayPal            31      1.0   
...          ...                    ...            ...           ...      ...   
249995         1                   3664           Cash            63      0.0   
249996         1                   4374    Credit Card            66      1.0   
249997         5                   5296           Cash            63      NaN   
249998         2                   2517           Cash            64      1.0   
249999         5                   3634           Cash            32      0.0   

            Customer Name  Age  Gender  Churn  
0             John Rivera   31  Female      0  
1             John Rivera   31  Female      0  
2             John Rivera   31  Female      0  
3             John Rivera   31  Female      0  
4             John Rivera   31  Female      0  
...                   ...  ...     ...    ...  
249995   Gabriel Williams   63    Male      0  
249996       Barry Foster   66  Female      0  
249997       Lisa Johnson   63  Female      0  
249998  Melissa Fernandez   64    Male      0  
249999      Angela Norton   32    Male      0  

[250000 rows x 13 columns]

In [57]:
data.describe()

Customer ID  Product Price       Quantity  Total Purchase Amount  \
count  250000.000000  250000.000000  250000.000000          250000.000000   
mean    25017.632092     254.742724       3.004936            2725.385196   
std     14412.515718     141.738104       1.414737            1442.576095   
min         1.000000      10.000000       1.000000             100.000000   
25%     12590.000000     132.000000       2.000000            1476.000000   
50%     25011.000000     255.000000       3.000000            2725.000000   
75%     37441.250000     377.000000       4.000000            3975.000000   
max     50000.000000     500.000000       5.000000            5350.000000   

        Customer Age        Returns            Age         Churn  
count  250000.000000  202618.000000  250000.000000  250000.00000  
mean       43.798276       0.500824      43.798276       0.20052  
std        15.364915       0.500001      15.364915       0.40039  
min        18.000000       0.000000      18.000000       0.00000  
25%        30.000000       0.000000      30.000000       0.00000  
50%        44.000000       1.000000      44.000000       0.00000  
75%        57.000000       1.000000      57.000000       0.00000  
max        70.000000       1.000000      70.000000       1.00000

Dari data di atas, diketahui bahwa harga produk termurah adalah 10 dolar, sedangkan harga termahal mencapai 500 dolar per produk. Rata-rata pembelian mencapai nilai 2725 dolar. Selain itu, usia rata-rata pembeli adalah 43 tahun, dengan jumlah barang yang dibeli per orang berkisar antara 3 hingga 5 barang, di mana 5 barang adalah jumlah tertinggi yang dibeli per orang.

In [58]:
data.columns

Index(['Customer ID', 'Purchase Date', 'Product Category', 'Product Price',
       'Quantity', 'Total Purchase Amount', 'Payment Method', 'Customer Age',
       'Returns', 'Customer Name', 'Age', 'Gender', 'Churn'],
      dtype='object')

# **Home**

In [59]:
data_home = data[data['Product Category'] == 'Home']
data_home

Customer ID        Purchase Date Product Category  Product Price  \
0             44605  2023-05-03 21:30:02             Home            177   
5             13738  2022-08-25 06:48:33             Home            191   
8             13738  2021-12-21 03:29:05             Home             12   
11            33969  2023-01-05 11:15:27             Home            304   
15            33969  2022-07-21 04:25:44             Home            193   
...             ...                  ...              ...            ...   
249983        41724  2020-08-23 23:53:47             Home            392   
249984        27527  2023-08-11 18:07:56             Home            294   
249990          150  2023-08-23 01:47:29             Home            196   
249995        33807  2023-01-24 12:32:18             Home            436   
249999         4148  2020-09-07 05:12:19             Home            307   

        Quantity  Total Purchase Amount Payment Method  Customer Age  Returns  \
0              1                   2427         PayPal            31      1.0   
5              3                   3722    Credit Card            27      1.0   
8              2                   2175           Cash            27      NaN   
11             1                   3883         PayPal            27      1.0   
15             2                   3198    Credit Card            27      0.0   
...          ...                    ...            ...           ...      ...   
249983         1                    863           Cash            55      1.0   
249984         1                    527    Credit Card            26      0.0   
249990         3                   1743         PayPal            68      0.0   
249995         1                   3664           Cash            63      0.0   
249999         5                   3634           Cash            32      0.0   

           Customer Name  Age  Gender  Churn  
0            John Rivera   31  Female      0  
5         Lauren Johnson   27  Female      0  
8         Lauren Johnson   27  Female      0  
11           Carol Allen   27    Male      0  
15           Carol Allen   27    Male      0  
...                  ...  ...     ...    ...  
249983       Katie Allen   55    Male      0  
249984        Ryan Marsh   26  Female      0  
249990  Stephanie Morris   68    Male      0  
249995  Gabriel Williams   63    Male      0  
249999     Angela Norton   32    Male      0  

[62542 rows x 13 columns]

In [60]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

data_home = data[data['Product Category'] == 'Home']

x = data_home[['Product Price','Quantity','Customer Age']] #Features
y = data_home[['Total Purchase Amount']] #target

#Split Data
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)

#Training
model = LinearRegression()
model.fit(x_train, y_train)

#Evaluasi
y_pred_train = model.predict(x_train)
y_pred = model.predict(x_test)

# Evaluasi Training
mse_train = mean_squared_error(y_pred_train,y_train)
print(f'Hasil Mean Squared Error Training = {mse_train:.3f}')

#Testing
mse = mean_squared_error(y_pred,y_test)
print(f'Hasil Mean Squared Error Testing = {mse:.3f}')

Hasil Mean Squared Error Training = 2081072.165
Hasil Mean Squared Error Testing = 2088249.487


hasil dari metode yang digunakan yaitu linear regresission menunjukkan bahwa hasil tersebut underfitting

In [61]:
import numpy as np
feature_importance = pd.DataFrame({
    'Feature' : x.columns,
    'Importance' : model.coef_[0]
})

feature_importance = feature_importance.sort_values(by = 'Importance', ascending= False)
print (feature_importance)

         Feature  Importance
2   Customer Age    5.514780
0  Product Price   -0.047602
1       Quantity   -2.876230


Penjualan total untuk produk kategori Home dipengaruhi oleh Customer Age

#**Electronics**

In [62]:
data_Electronics = data[data['Product Category'] == 'Electronics']
data_Electronics

Customer ID        Purchase Date Product Category  Product Price  \
1             44605  2021-05-16 13:57:44      Electronics            174   
3             44605  2023-01-17 13:14:36      Electronics            396   
6             13738  2023-07-25 05:17:24      Electronics            205   
9             13738  2023-02-09 00:53:14      Electronics             40   
13            33969  2021-12-20 23:44:57      Electronics            428   
...             ...                  ...              ...            ...   
249993        39806  2021-12-07 05:42:38      Electronics            215   
249994        39806  2021-08-01 04:43:12      Electronics            225   
249996        20455  2021-06-04 05:45:25      Electronics            233   
249997        28055  2022-11-10 17:11:57      Electronics            441   
249998        15023  2021-06-27 14:42:12      Electronics             44   

        Quantity  Total Purchase Amount Payment Method  Customer Age  Returns  \
1              3                   2448         PayPal            31      1.0   
3              3                    937           Cash            31      0.0   
6              1                   2773    Credit Card            27      NaN   
9              4                   4327           Cash            27      0.0   
13             4                   2289           Cash            27      0.0   
...          ...                    ...            ...           ...      ...   
249993         3                   2349           Cash            60      1.0   
249994         5                   5293    Credit Card            60      0.0   
249996         1                   4374    Credit Card            66      1.0   
249997         5                   5296           Cash            63      NaN   
249998         2                   2517           Cash            64      1.0   

            Customer Name  Age  Gender  Churn  
1             John Rivera   31  Female      0  
3             John Rivera   31  Female      0  
6          Lauren Johnson   27  Female      0  
9          Lauren Johnson   27  Female      0  
13            Carol Allen   27    Male      0  
...                   ...  ...     ...    ...  
249993         Dana Brown   60  Female      0  
249994         Dana Brown   60  Female      0  
249996       Barry Foster   66  Female      0  
249997       Lisa Johnson   63  Female      0  
249998  Melissa Fernandez   64    Male      0  

[62630 rows x 13 columns]

In [63]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

data_Electronics = data[data['Product Category'] == 'Electronics']

x = data_Electronics[['Product Price', 'Quantity', 'Customer Age']] #Features
y = data_Electronics[['Total Purchase Amount']] #target

#Split Data
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)

#Training
model = LinearRegression()
model.fit(x_train, y_train)

#Evaluasi
y_pred = model.predict(x_test)

mse = mean_squared_error(y_train,y_train)
print(f'Hasil Menn Squared Error training = {mse:.3f}')

#Testing
mse = mean_squared_error(y_pred,y_test)
print(f'Hasil Mean Squared Error Testing = {mse:.3f}')

Hasil Menn Squared Error training = 0.000
Hasil Mean Squared Error Testing = 2074028.724



Hasil MSE pada Electronics training = 0 menunjukkan model sangat cocok dengan data pelatihan, namun MSE testing yang tinggi yang membuat model gagal memprediksi data baru. Ini menandakan adanya overfitting

In [64]:
import numpy as np
feature_importance = pd.DataFrame({
    'Feature' : x.columns,
    'Importance' : model.coef_[0]
})

feature_importance = feature_importance.sort_values(by = 'Importance', ascending= False)
print (feature_importance)

         Feature  Importance
2   Customer Age    5.469367
0  Product Price   -0.005988
1       Quantity   -5.879287


Penjualan total untuk produk kategori Electronics dipengaruhi oleh Customer Age

#**Clothing**

In [65]:
data_Clothing = data[data['Product Category'] == 'Clothing']
data_Clothing

Customer ID        Purchase Date Product Category  Product Price  \
10            33969  2023-02-28 19:58:23         Clothing            410   
16            33969  2023-07-05 15:01:04         Clothing            473   
20            42650  2021-04-21 07:58:37         Clothing            252   
35            19676  2022-01-12 22:49:28         Clothing            304   
37            19676  2022-06-30 06:11:14         Clothing            306   
...             ...                  ...              ...            ...   
249978        18409  2021-05-24 10:23:27         Clothing            490   
249980        39396  2020-03-08 08:01:57         Clothing            147   
249985         1887  2023-03-31 07:16:20         Clothing            500   
249987         8255  2022-05-02 06:57:17         Clothing            139   
249988        31150  2022-03-25 11:15:14         Clothing            101   

        Quantity  Total Purchase Amount Payment Method  Customer Age  Returns  \
10             3                   5018    Credit Card            27      NaN   
16             3                   2881    Credit Card            27      1.0   
20             2                   1775    Credit Card            20      1.0   
35             3                   4869         PayPal            57      1.0   
37             4                   1712    Credit Card            57      1.0   
...          ...                    ...            ...           ...      ...   
249978         5                   2883           Cash            64      0.0   
249980         1                   1737    Credit Card            54      NaN   
249985         3                   3268         PayPal            30      NaN   
249987         2                   5274         PayPal            61      1.0   
249988         5                   2387    Credit Card            26      NaN   

             Customer Name  Age  Gender  Churn  
10             Carol Allen   27    Male      0  
16             Carol Allen   27    Male      0  
20            Curtis Smith   20  Female      0  
35               Linda Lee   57    Male      0  
37               Linda Lee   57    Male      0  
...                    ...  ...     ...    ...  
249978      Carrie Summers   64    Male      0  
249980    Melissa Williams   54  Female      0  
249985  Christina Alvarado   30  Female      0  
249987      Jasmine Adkins   61  Female      0  
249988      Jennifer Mason   26    Male      0  

[62581 rows x 13 columns]

In [66]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

data_Clothing = data[data['Product Category'] == 'Clothing']

x = data_Clothing[['Product Price', 'Quantity', 'Customer Age']] #Features
y = data_Clothing[['Total Purchase Amount']] #target

#Split Data
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)

#Training
model = LinearRegression()
model.fit(x_train, y_train)

#Evaluasi
y_pred = model.predict(x_test)

mse = mean_squared_error(y_train,y_train)
print(f'Hasil Menn Squared Error training = {mse:.3f}')

#Testing
mse = mean_squared_error(y_pred,y_test)
print(f'Hasil Mean Squared Error Testing = {mse:.3f}')

Hasil Menn Squared Error training = 0.000
Hasil Mean Squared Error Testing = 2076500.782



Hasil MSE pada Clothing training = 0 menunjukkan model sangat cocok dengan data pelatihan, namun MSE testing yang tinggi yang membuat model gagal memprediksi data baru. Ini menandakan adanya overfitting

In [67]:
import numpy as np
feature_importance = pd.DataFrame({
    'Feature' : x.columns,
    'Importance' : model.coef_[0]
})

feature_importance = feature_importance.sort_values(by = 'Importance', ascending= False)
print (feature_importance)

         Feature  Importance
2   Customer Age    5.688819
1       Quantity    5.369057
0  Product Price   -0.058414


Penjualan total untuk produk kategori Clothing dipengaruhi oleh Customer Age

#**Data Laki Laki umur di atas rata-rata**

In [68]:
data.columns

Index(['Customer ID', 'Purchase Date', 'Product Category', 'Product Price',
       'Quantity', 'Total Purchase Amount', 'Payment Method', 'Customer Age',
       'Returns', 'Customer Name', 'Age', 'Gender', 'Churn'],
      dtype='object')

In [69]:
data['Customer Age'].mean()

43.798276

In [70]:
data_laki2 = data[data['Gender'] == 'Male']
data_laki2

Customer ID        Purchase Date Product Category  Product Price  \
10            33969  2023-02-28 19:58:23         Clothing            410   
11            33969  2023-01-05 11:15:27             Home            304   
12            33969  2023-07-18 23:36:50            Books             54   
13            33969  2021-12-20 23:44:57      Electronics            428   
14            33969  2020-03-07 21:31:35            Books            281   
...             ...                  ...              ...            ...   
249991        16247  2020-06-24 17:29:27      Electronics            368   
249992        48538  2020-12-13 02:09:15      Electronics            239   
249995        33807  2023-01-24 12:32:18             Home            436   
249998        15023  2021-06-27 14:42:12      Electronics             44   
249999         4148  2020-09-07 05:12:19             Home            307   

        Quantity  Total Purchase Amount Payment Method  Customer Age  Returns  \
10             3                   5018    Credit Card            27      NaN   
11             1                   3883         PayPal            27      1.0   
12             2                   4187         PayPal            27      0.0   
13             4                   2289           Cash            27      0.0   
14             1                   3810           Cash            27      0.0   
...          ...                    ...            ...           ...      ...   
249991         1                   1720    Credit Card            22      0.0   
249992         1                   2356           Cash            61      1.0   
249995         1                   3664           Cash            63      0.0   
249998         2                   2517           Cash            64      1.0   
249999         5                   3634           Cash            32      0.0   

            Customer Name  Age Gender  Churn  
10            Carol Allen   27   Male      0  
11            Carol Allen   27   Male      0  
12            Carol Allen   27   Male      0  
13            Carol Allen   27   Male      0  
14            Carol Allen   27   Male      0  
...                   ...  ...    ...    ...  
249991      Mike Thompson   22   Male      0  
249992      Alexis Nelson   61   Male      1  
249995   Gabriel Williams   63   Male      0  
249998  Melissa Fernandez   64   Male      0  
249999      Angela Norton   32   Male      0  

[125676 rows x 13 columns]

In [71]:
data_laki2 = data[data['Gender'] == 'Male']
data_laki2_mean = data_laki2[data_laki2['Customer Age'] > data_laki2['Customer Age'].mean()]

In [72]:
data_laki2 = data[data['Gender'] == 'Male']
data_laki2_mean = data_laki2[data_laki2['Customer Age'] > data_laki2['Customer Age'].mean()]

x = data_laki2_mean[['Product Price','Quantity','Customer Age']] #Features
y = data_laki2_mean[['Total Purchase Amount']] #target

#Split Data
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)

#Training
model = LinearRegression()
model.fit(x_train, y_train)

#Evaluasi
y_pred = model.predict(x_test)

mse = mean_squared_error(y_pred,y_test)
print(f'Hasil Mean Squared Error = {mse:.3f}')

Hasil Mean Squared Error = 2065957.425


Hasil Mean Squared Error (MSE) = 2065957.425 pada data laki-laki dengan usia rata-rata di atas menunjukkan bahwa model mungkin mengalami underfitting

In [73]:
import numpy as np
feature_importance = pd.DataFrame({
    'Feature': x.columns,
    'Importance': model.coef_[0]
})

feature_importance = feature_importance.sort_values(by='Importance',ascending= False)
print(feature_importance)

         Feature  Importance
2   Customer Age    5.435304
1       Quantity    2.438187
0  Product Price   -0.042176


#**Data Perempuan umur di atas rata-rata**

In [74]:
data_perempuan = data[data['Gender'] == 'Female']
data_perempuan

Customer ID        Purchase Date Product Category  Product Price  \
0             44605  2023-05-03 21:30:02             Home            177   
1             44605  2021-05-16 13:57:44      Electronics            174   
2             44605  2020-07-13 06:16:57            Books            413   
3             44605  2023-01-17 13:14:36      Electronics            396   
4             44605  2021-05-01 11:29:27            Books            259   
...             ...                  ...              ...            ...   
249987         8255  2022-05-02 06:57:17         Clothing            139   
249993        39806  2021-12-07 05:42:38      Electronics            215   
249994        39806  2021-08-01 04:43:12      Electronics            225   
249996        20455  2021-06-04 05:45:25      Electronics            233   
249997        28055  2022-11-10 17:11:57      Electronics            441   

        Quantity  Total Purchase Amount Payment Method  Customer Age  Returns  \
0              1                   2427         PayPal            31      1.0   
1              3                   2448         PayPal            31      1.0   
2              1                   2345    Credit Card            31      1.0   
3              3                    937           Cash            31      0.0   
4              4                   2598         PayPal            31      1.0   
...          ...                    ...            ...           ...      ...   
249987         2                   5274         PayPal            61      1.0   
249993         3                   2349           Cash            60      1.0   
249994         5                   5293    Credit Card            60      0.0   
249996         1                   4374    Credit Card            66      1.0   
249997         5                   5296           Cash            63      NaN   

         Customer Name  Age  Gender  Churn  
0          John Rivera   31  Female      0  
1          John Rivera   31  Female      0  
2          John Rivera   31  Female      0  
3          John Rivera   31  Female      0  
4          John Rivera   31  Female      0  
...                ...  ...     ...    ...  
249987  Jasmine Adkins   61  Female      0  
249993      Dana Brown   60  Female      0  
249994      Dana Brown   60  Female      0  
249996    Barry Foster   66  Female      0  
249997    Lisa Johnson   63  Female      0  

[124324 rows x 13 columns]

In [75]:
data_perempuan = data[data['Gender'] == 'Female']
data_perempuan_mean = data_perempuan[data_perempuan['Customer Age'] > data_perempuan['Customer Age'].mean()]

In [76]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

data_perempuan = data[data['Gender'] == 'Female']
data_perempuan_mean = data_perempuan[data_perempuan['Customer Age'] > data_perempuan['Customer Age'].mean()]

x = data_perempuan_mean[['Product Price', 'Quantity', 'Customer Age']] #Features
y = data_perempuan_mean[['Total Purchase Amount']] #target

#Split Data
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.2, random_state = 0)

#Training
model = LinearRegression()
model.fit(x_train, y_train)

#Evaluasi
y_pred = model.predict(x_test)

mse = mean_squared_error(y_pred,y_test)
print(f'Hasil Menan Squared Error = {mse:.3f}')

Hasil Menan Squared Error = 2069618.737


Hasil Mean Squared Error (MSE) = 2069618.737 pada data perempuan dengan usia rata-rata di atas menunjukkan bahwa model mungkin mengalami overfitting

In [77]:
import numpy as np
feature_importance = pd.DataFrame({
    'Feature': x.columns,
    'Importance': model.coef_[0]
})

feature_importance = feature_importance.sort_values(by='Importance',ascending= False)
print(feature_importance)

         Feature  Importance
2   Customer Age    5.119484
0  Product Price    0.023025
1       Quantity   -1.971949



Dari hasil feature importance, Customer Age memiliki pengaruh yang lebih besar pada data laki-laki dengan nilai 5.435304, dibandingkan dengan perempuan yang memiliki nilai 5.119484. Jadi tinggi atau banyak pada laki-laki

In [78]:
data_Credit_Card = data[data['Payment Method'] == 'Credit Card']
data_Credit_Card.describe()

Customer ID  Product Price      Quantity  Total Purchase Amount  \
count  83547.000000   83547.000000  83547.000000           83547.000000   
mean   25024.098148     254.324943      3.004153            2738.852562   
std    14427.915187     141.681873      1.415523            1444.037283   
min        1.000000      10.000000      1.000000             100.000000   
25%    12558.500000     131.000000      2.000000            1488.000000   
50%    25014.000000     254.000000      3.000000            2747.000000   
75%    37501.000000     376.000000      4.000000            3990.000000   
max    50000.000000     500.000000      5.000000            5349.000000   

       Customer Age       Returns           Age         Churn  
count  83547.000000  67517.000000  83547.000000  83547.000000  
mean      43.861336      0.501474     43.861336      0.196620  
std       15.364194      0.500002     15.364194      0.397445  
min       18.000000      0.000000     18.000000      0.000000  
25%       30.000000      0.000000     30.000000      0.000000  
50%       44.000000      1.000000     44.000000      0.000000  
75%       57.000000      1.000000     57.000000      0.000000  
max       70.000000      1.000000     70.000000      1.000000

In [79]:
data_Paypal = data[data['Payment Method'] == 'PayPal']
data_Paypal.describe()

Customer ID  Product Price      Quantity  Total Purchase Amount  \
count  83441.000000   83441.000000  83441.000000           83441.000000   
mean   25039.936482     255.288851      3.012416            2721.677952   
std    14400.641348     141.705453      1.412395            1441.397955   
min        2.000000      10.000000      1.000000             104.000000   
25%    12625.000000     133.000000      2.000000            1471.000000   
50%    25074.000000     256.000000      3.000000            2724.000000   
75%    37407.000000     377.000000      4.000000            3970.000000   
max    50000.000000     500.000000      5.000000            5344.000000   

       Customer Age       Returns           Age         Churn  
count  83441.000000  67811.000000  83441.000000  83441.000000  
mean      43.708261      0.500391     43.708261      0.202430  
std       15.361336      0.500004     15.361336      0.401814  
min       18.000000      0.000000     18.000000      0.000000  
25%       30.000000      0.000000     30.000000      0.000000  
50%       44.000000      1.000000     44.000000      0.000000  
75%       57.000000      1.000000     57.000000      0.000000  
max       70.000000      1.000000     70.000000      1.000000

Tidak memerlukan saran, karena total pengguna yang menggunakan PayPal adalah 83.441, namun kenyataannya penggunaan kartu kredit jauh lebih banyak dibandingkan dengan PayPal.
